samples/gesture_detector/src/sample/gesture_detector.cljd

```diff
@@ -0,0 +1,77 @@
+(ns sample.gesture-detector
+  (:require
+   ["package:flutter/gestures.dart" :as g]
+   ["package:flutter/material.dart" :as m]
+   ["package:vector_math/vector_math_64.dart" :as vm]
+   [cljd.flutter.alpha :as f]))
+
+(def radius 25)
+
+(defn inside-parallelogram
+  [[o a b] mouse-pos]
+  (let [oa (.- a o)
+        ob (.- b o)
+        m (doto
+           (vm/Matrix3 (.-dx oa) (.-dy oa) 0
+                      (.-dx ob) (.-dy ob) 0
+                      (.-dx o)  (.-dy o)  1)
+            .invert) 
+        p-1 (.* m (vm/Vector3 (.-dx mouse-pos) (.-dy mouse-pos) 1))]
+    (dart:core/print (str p-1))
+    (when (and (<= 0 (.-x p-1) 1) (<= 0 (.-y p-1) 1))
+      (.- mouse-pos o))))
+
+(defn move-to
+  [[o a b] o']
+  (let [oo' (.- o' o)]
+    [o' (.+ a oo') (.+ b oo')]))
+
+
+(def gesture-parallelogram
+  (f/widget
+   :state [app-state {:vertices [(m/Offset 150 150) (m/Offset 250 150) (m/Offset 150 350)]
+                      :delta nil :dragged nil}]
+   (m/GestureDetector
+    :onPanStart
+    (fn [^g/DragStartDetails details]
+      (when-some [delta (inside-parallelogram (:vertices @app-state) (.localPosition details))]
```

Comment:
While it works here, it's considered bad style to deref the same atom several times.
That's because in more concurrent env (eg Clojure or even cld/cljs with an async op between the derefs) deref may not always return the same value thus leading to inconsistent behaviors.
To a lesser extent multiple swap! are dubious.

So here the deref and the swap! should be pushed to the ends of the method and the middle should be purely functional. (This is a general trend in Clojure: function core, imperative shell)